probe/endpoint/ebpf.go

```diff
@@ -7,36 +7,29 @@ import (
 	"os/exec"
 	"strconv"
 	"strings"
+	"sync"
 
 	log "github.com/Sirupsen/logrus"
 )
 
-type event int
-
-const (
-	// Connect is a TCP CONNECT event
-	Connect event = iota
-	// Accept is a TCP ACCEPT event
-	Accept
-	// Close is a TCP CLOSE event
-	Close
-)
-
-// A ConnectionEvent represents a network connection
-type ConnectionEvent struct {
-	Type          event
-	Pid           int
-	Command       string
-	SourceAddress net.IP
-	DestAddress   net.IP
-	SourcePort    uint16
-	DestPort      uint16
+// A ebpfConnection represents a network connection
+type ebpfConnection struct {
+	tuple    fourTuple
+	netns    string
+	outgoing bool
+	pid      int
```

Comment:
Maybe, I don't know. It would be logical to me but I see this pid_t is signed:
http://lxr.free-electrons.com/source/include/uapi/asm-generic/posix_types.h#L27